android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/MainActivity.java

```diff
@@ -517,7 +517,7 @@ private void downloadKMP(String scheme) {
           if (progressDialog == null) {
             progressDialog = new ProgressDialog(MainActivity.this);
             progressDialog.setMessage(String.format(getString(R.string.downloading_keyboard_package), filename));
-            progressDialog.setCancelable(false);
+            progressDialog.setCancelable(true); // Cancelable in case there's exceptions
```

Comment:
Question - what happens if the user cancels it while displayed and operating successfully?

Reply:
Is there any way to adjust this value after the fact if/when we detect an exception?

Reply:
> Question - what happens if the user cancels it while displayed and operating successfully?

The download dialog is hidden by the package installer until it finishes (readme.htm -> language picker -> welcome.htm)

Reply:
Is it also displayed before the package installer launches?  What happens if the user cancels due to poor internet connection, but the initial package's download then succeeds?

Reply:
Yes, the Download dialog is shown before the package installer launches.
I tested what happens if the user cancels the "Download dialog", and the .kmp still downloads.

The .kmp file launches the package installer which can proceed as normally